test.nix

```diff
@@ -104,7 +104,7 @@ let
           npins
           python3
           netcat
-          nix
+          lix
```

Comment:
I'm fine with using `lix` for this but we should support running tests against `nix` as well. I don't think we have to default to brokenNix (better name than `CppNix` IMO) but there might be users and perhaps at some point we can add a smoke test as even `lix` might introduce bugs at some point.

Reply:
Not sure how to best do this. `npins` has its `nix` dependency fixed in the derivation because of `wrapProgram $out/bin/npins --prefix PATH : "${runtimePath}"`, so even if the test fixture is flexible it still doesn't change what nix implementation npins uses for prefetching. Maybe we should just remove that and make it a "classic" run-time dependencey?

Reply:
Maybe, I think in the past that might have helped keep the nix biniaries compatible with what we test against but ideally we'd use the fetching code of `tvix` anyway (or implement it on our own).